### ISSUE TITLE
Running both TensorFlow and gym on Apple Silicon

### ISSUE THREAD

**Issue by Ryan-Rudes** [openai/gym#2134]
I have been able to follow the workarounds to install Apple's special TensorFlow distribution on the M1 Mac Mini. I have also been able to successfully install Open AI's gym on the device. However, due to the procedure of installing Apple's special TensorFlow distribution for Mac Silicon, I don't seem to be apple to run Apple's TensorFlow alongside some packages like Pandas or Scipy. As Scipy is a requirement for Gym, I have not yet figured out how to install TensorFlow and Gym to be used together on Mac Silicon. Are there any tips or potentially workaround for getting this to work?

**Comment by greamy** [comment]
i have been working on this too, and am at a loss for where to go next. It seems installing scipy is a challenge using conventional methods like pip

**Comment by Ryan-Rudes** [comment]
@greamy, Since I raised this question many months ago, I've since resolved most of my problems with Mac M1 dependency issues (except those which cannot be fixed until compatible software is released). For scipy, you can simply install it with miniforge. For installing tensorflow, follow [this](https://github.com/apple/tensorflow_macos/issues/153) tutorial (and make sure to very carefully follow all instructions). Only certain builds of Python combined with certain package managers will enable everything to work nicely. On my machine, I am currently able to run TensorFlow, cv2 (you can install this with conda, too), PyTorch, Pygame, OpenAI Gym, Scipy, Scikit-learn, and many other data-science packages under one Python. I hope this helps.

**Comment by mwidjaja1** [comment]
Hey there! If you have any questions while following that tutorial, feel free to ask. I wrote it, I _know_ it's a doozy, but it is a doozy process so I'd be glad to help as much as I can virtually.

I'm not an official Apple employee so I'm speculating, but my guess is that they wouldn't formally merge their M1 TensorFlow updates into master until they know it's working solidly and using M1 to its maximum potential. I'm sure Apple, like most corporations, have incredibly strict processes on sharing their work with the greater community, and the legalities of sharing it with TF formally is tighter than Apple's GitLab. My guess is that a formal merge will happen, it'll just happen after the Apple Silicon transition for all of their Macs wrap up.
